src/gt4py/next/iterator/embedded.py

```diff
@@ -878,60 +888,53 @@ def make_in_iterator(
+
+    _ndarrayfield: common.Field
 
     @property
     def __gt_dims__(self) -> tuple[common.Dimension, ...]:
-        return self._axes
-
-    def __init__(
-        self,
-        getter: Callable[[FieldIndexOrIndices], Any],
-        axes: tuple[common.Dimension, ...],
-        dtype,
-        *,
-        setter: Callable[[FieldIndexOrIndices, Any], None],
-        array: Callable[[], npt.NDArray],
-        origin: Optional[dict[common.Dimension, int]] = None,
-    ):
-        self.getter = getter
-        self._axes = axes
-        self.setter = setter
-        self.array = array
-        self.dtype = dtype
-        self.origin = origin
-
-    def __getitem__(self, indices: ArrayIndexOrIndices) -> Any:
-        return self.array()[indices]
+        return self._ndarrayfield.__gt_dims__
+
+    def _promote_tags_to_dims_and_sort(
+        self, named_indices: NamedFieldIndices
+    ) -> Mapping[common.Dimension, FieldIndex | SparsePositionEntry]:
+        return {d: named_indices[d.value] for d in self._ndarrayfield.__gt_dims__}
+
+    def _translate_named_indices(self, _named_indices: NamedFieldIndices) -> common.DomainSlice:
+        named_indices = self._promote_tags_to_dims_and_sort(_named_indices)
+        domain_slice: list[common.NamedRange | common.NamedIndex] = []
+        for d, v in named_indices.items():
+            if isinstance(v, range):
+                domain_slice.append((d, common.UnitRange(v.start, v.stop)))
+            elif isinstance(v, list):
+                assert len(v) == 1  # only 1 sparse dimension is supported
+                assert common.is_int_index(
+                    v[0]
+                )  # derefing a concrete element in a sparse field, not a slice
+                domain_slice.append((d, v[0]))
+            else:
+                assert common.is_int_index(v)
+                domain_slice.append((d, v))
+        return tuple(domain_slice)
 
-    # TODO in a stable implementation of the Field concept we should make this behavior the default behavior for __getitem__
     def field_getitem(self, named_indices: NamedFieldIndices) -> Any:
-        return self.getter(get_ordered_indices(self._axes, named_indices))
-
-    def __setitem__(self, indices: ArrayIndexOrIndices, value: Any):
-        self.array()[indices] = value
+        return self._ndarrayfield[self._translate_named_indices(named_indices)]
 
     def field_setitem(self, named_indices: NamedFieldIndices, value: Any):
-        self.setter(get_ordered_indices(self._axes, named_indices), value)
+        if common.is_mutable_field(self._ndarrayfield):
+            self._ndarrayfield[self._translate_named_indices(named_indices)] = value
+        else:
+            raise RuntimeError("Assigment into a non-mutable Field.")
 
-    def __array__(self) -> np.ndarray:
-        return self.array()
+    def __array__(self) -> core_defs.NDArrayObject:
```

Comment:
Incomplete signature (missing dtype argument)

Reply:
Trying to delete it, not sure if it's actually used.